File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/model/PartitionDependencyBuilder.java
package io.qimia.uhrwerk.common.model;

public class PartitionDependencyBuilder {

  Long id;
  Long partitionId;
  Long dependencyPartitionId;

  public PartitionDependencyBuilder id(Long id) {
    this.id = id;
    return this;
  }

  public PartitionDependencyBuilder partitionId(Long partitionId) {
    this.partitionId = partitionId;
    return this;
  }

  public PartitionDependencyBuilder dependencyPartitionId(Long dependencyPartitionId) {
    this.dependencyPartitionId = dependencyPartitionId;
    return this;
  }

  public PartitionDependency build() {
    return new PartitionDependency(this);
  }
}

File: uhrwerk-metastore/src/main/kotlin/io/qimia/uhrwerk/common/metastore/builders/PartitionDependencyBuilder.kt
package io.qimia.uhrwerk.common.metastore.builders

import io.qimia.uhrwerk.common.metastore.model.PartitionDependency

class PartitionDependencyBuilder {
    var id: Long? = null
    var partitionId: Long? = null
    var dependencyPartitionId: Long? = null
    fun id(id: Long?): PartitionDependencyBuilder {
        this.id = id
        return this
    }

    fun partitionId(partitionId: Long?): PartitionDependencyBuilder {
        this.partitionId = partitionId
        return this
    }

    fun dependencyPartitionId(dependencyPartitionId: Long?): PartitionDependencyBuilder {
        this.dependencyPartitionId = dependencyPartitionId
        return this
    }

    fun build(): PartitionDependency {
        return PartitionDependency(id, partitionId, dependencyPartitionId)
    }
}